applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_isotropic_plasticity.cpp

```diff
@@ -128,8 +119,10 @@ void GenericSmallStrainIsotropicPlasticity<TConstLawIntegratorType>::CalculateMa
             if( r_constitutive_law_options.Is( ConstitutiveLaw::U_P_LAW ) ) {
                 predictive_stress_vector = rValues.GetStressVector();
             } else {
-                // S0 = r_constitutive_matrix:(E-Ep)
-                predictive_stress_vector = prod(r_constitutive_matrix, r_strain_vector - plastic_strain);
+                // S0 = Elastic stress with strain (E-Ep)

```

Comment:
why the case of U_P_LAW is different?

Reply:
@AlejandroCornejo already explained this